include/messages.h

```diff
@@ -0,0 +1,552 @@
+#ifndef __BM_MESSAGES_H__
+#define __BM_MESSAGES_H__
+
+#include <stdbool.h>
+#include <stdint.h>
+
+typedef struct {
+  uint16_t type;
+  uint16_t checksum;
+  uint8_t flags;
+  uint8_t reserved;
+  uint32_t seq_num;
+  uint8_t frag_total;
+  uint8_t frag_id;
+  uint8_t next_header;
+} __attribute__((packed)) BCMPHeader;
+#define bcmp_header_struct_layout \
+  BCMP_PARSER_16BIT, \
+  BCMP_PARSER_16BIT, \
+  BCMP_PARSER_8BIT, \
+  BCMP_PARSER_8BIT, \
+  BCMP_PARSER_32BIT, \
+  BCMP_PARSER_8BIT, \
+  BCMP_PARSER_8BIT, \
+  BCMP_PARSER_8BIT,
+#define bcmp_header_serialize_params { {bcmp_header_struct_layout}, sizeof(BCMPHeader) }
```

Comment:
Essentially every new message added here will have these associated macros with them that will layout the struct in association with the `bcmp_serialize` function.
There is a helper macro `bcmp_serialize_data` that can be used to easily integrate the `_params` macro into tx/rx messages.